shared/actions/engine/helper.js

```diff
@@ -152,6 +160,11 @@ class EngineRpcCall {
         }
 
         if (incoming.finished) {
+          // Used just by device add for now. this is to fix a bug and i'm not sure this should apply generally
+          if (incoming.finished.error && this._finishedErrorShouldThrow) {
```

Comment:
I'm not sure why this didn't work previously.  Seems like `yield call([this, this._cleanup], subSagaTasks)` would still be called (Eeven if an error) and then control would resume in the saga.

Reply:
before it'd just wait for subsagas to finish themselves and wouldn't interrupt them